Ci dependencies

paulr34 opened the pull request:
upgrading electron, 
deleting extra canvas call that isn't used and breaks everything,
upgrading quasar,
adding python tools to the release
and upgrading cypress

brdandu commented on cypress/e2e/attributes/check-cluster-attributes-singleton-in-zigbee-mode.cy.js.skip:
```diff
@@ -1,6 +1,6 @@
 /// <reference types="cypress" />
 
-Cypress.on('uncaught:exception', (err, runnable) => {
+Cypress.off('uncaught:exception', (err, runnable) => {
```
Why is this switched off?

paulr34 replied:
that is a typo, thank you

brdandu commented on cypress/e2e/clusters/cluster-multiple-search.cy.js:
```diff
@@ -49,10 +49,8 @@ describe('Add multiple clusters and search', () => {
   })
   it('Search action', () => {
     cy.fixture('data').then((data) => {
-      cy.get(
```
Can you get rid of this diff?

paulr34 replied:
yes, no problem , thank you

brdandu commented on cypress/e2e/custom_xml/custom_xml.cy.js.skip:
```diff
@@ -17,7 +17,7 @@ describe('Check ZCL Extension functionalities', () => {
       .then(function (window) {
         window[rendApi.GLOBAL_SYMBOL_EXECUTE](
-          rendApi.id.open,
+          restApi.ide.open,
```
what is the difference between id and ide?

brdandu replied:
ok

brdandu commented on .github/workflows/release.yml:
```diff
@@ -160,6 +160,8 @@ jobs:
           node-version: 18.x
           cache: 'npm'
+      - name: pip
+        run: pip install setuptools
```
What does setup tools do?

paulr34 replied:
So there were a handful of problems that this PR was solving. One of them being that we couldn't build a Windows or a Mac binary anymore. The Mac problem was due to dependency issues. Basically npm install would fail on a Mac before it even got to building a binary.

After the dependency issues were resolved, Mac binary built fine but Windows still had failures. One of the errors was that the distutils (dist folder) python modules had been "end of life". Basically, an alternative to using distutils was to install setuptools. After installing setuptools the Windows binary built.

https://www.geeksforgeeks.org/differences-between-distribute-distutils-setuptools-in-python/

brdandu replied:
Does this information need to be updated to any of our documentation for ZAP's source installation ?

paulr34 replied:
yes, probably

brdandu commented on cypress.config.ts:
Why is this a typescript file instead of a js file?

paulr34 replied:
This file was auto generated by the Cypress upgrade.

brdandu replied:
Is this essentially the same content as what was in `cypress.json`. Could this be put in a js file instead and it automatically works?

paulr34 replied:
where? Im looking in VScode and don't see it. I didn't manually add that file.

brdandu replied:
This one replaced https://github.com/project-chip/zap/blob/ca7937a6aebf3783fee548390fe35e30dc45be55/cypress.json

paulr34 commented on cypress/e2e/attribute_validations/bitmap.cy.js:
Cypress update which had to be done. Basically, upgrading something like Electron makes a new dependency tree and the old dependencies are not valid anymore.

The Cypress upgrade required a program to run and change the file extensions and some of the content.

More details here: https://docs.cypress.io/guides/references/configuration

brdandu replied:
Were all these files renamed automatically through some script or you had to figure it out file by file?

paulr34 replied:
script: https://docs.cypress.io/guides/references/configuration

brdandu reviewed (approved):
Approving given the CI passes like it should

tecimovic commented:
Doing an update is a good thing.

2 comments:
   1.) Run `npm audit`. See a lot of packages that are in various states of "security problem". Can we update many of those?
   2.) Run `npm outdated`. Long list. Can we updated many of those?
   3.) I see electron-builder is outdated. If we're upgrading electron, we should probably also update electron-builder. They go hand in hand.

paulr34 commented:
> Doing an update is a good thing.
> 
> 2 comments: 1.) Run `npm audit`. See a lot of packages that are in various states of "security problem". Can we update many of those? 2.) Run `npm outdated`. Long list. Can we updated many of those? 3.) I see electron-builder is outdated. If we're upgrading electron, we should probably also update electron-builder. They go hand in hand.

Thanks @tecimovic I updated electron-builder. 

1. audit causes failures so I will have to go through them 1 by one. This should be done in a follow up PR.
2.) npm outdated provided some good information and it should be done but in a "phase 2" second PR

agree?
